plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiSmokeTest.java

```diff
@@ -469,17 +510,156 @@ public void testPartitionFilterRequiredOnJoinBothTablePartitioned()
+        assertThat(matcher.find())
+                .withFailMessage("Could not find 'ScanFilter' for table '%s' with 'dynamicFilters' and 'Input: X rows' stats in EXPLAIN output.\nOutput was:\n%s",
+                        tableIdentifier, explainRes.toString())
+                .isTrue();
+
+        // matcher#group() must be invoked after matcher#find()
+        String rowsInputString = matcher.group(1);
+        long actualInputRows = Long.parseLong(rowsInputString);
+        long expectedInputRowsAfterFiltering = 2;
+        assertThat(actualInputRows)
+                .describedAs("Number of rows input to the ScanFilter for the probe side table (%s) should reflect effective dynamic filtering", tableIdentifier)
+                .isEqualTo(expectedInputRowsAfterFiltering);
+
+        // Exercise query and check output
+        assertQuery(query, "SELECT * FROM VALUES (1, 'a1', 100.0, 1000), (3, 'a3', 101.0, 1001)");
+    }
+
+    @Test
+    public void testDynamicFilterDisabledPredicatePushdown()
+    {
+        Session session = SessionPropertyConfigurator.from(getSession())
+                .withMdtEnabled(false)
+                .withDynamicFilterTimeout("0s")
+                .build();
+        final String tableIdentifier = "hudi:tests.hudi_multi_fg_pt_mor";
+        @Language("SQL") String query = "SELECT t1.id, t1.name, t1.price, t1.ts FROM " +
+                HUDI_MULTI_FG_PT_MOR + " t1 " +
+                "INNER JOIN " + HUDI_MULTI_FG_PT_MOR + " t2 ON t1.id = t2.id " +
+                "WHERE t2.price <= 102";
+
+        MaterializedResult explainRes = getQueryRunner().execute(session, "EXPLAIN ANALYZE " + query);
+        Pattern scanFilterInputRowsPattern = getScanFilterInputRowsPattern(tableIdentifier);
+        Matcher matcher = scanFilterInputRowsPattern.matcher(explainRes.toString());
+        assertThat(matcher.find())
+                .withFailMessage("Could not find 'ScanFilter' for table '%s' with 'dynamicFilters' and 'Input: X rows' stats in EXPLAIN output.\nOutput was:\n%s",
+                        tableIdentifier, explainRes.toString())
+                .isTrue();
+
+        // matcher#group() must be invoked after matcher#find()
+        String rowsInputString = matcher.group(1);
+        long actualInputRows = Long.parseLong(rowsInputString);
+        // No dynamic filtering performed, all rows read
+        long expectedInputRowsAfterFiltering = 4;
+        assertThat(actualInputRows)
+                .describedAs("Number of rows input to the ScanFilter for the probe side table (%s) should NOT reflect effective dynamic filtering", tableIdentifier)
+                .isEqualTo(expectedInputRowsAfterFiltering);
+
+        // Exercise query and check output
+        assertQuery(query, "SELECT * FROM VALUES (1, 'a1', 100.0, 1000), (3, 'a3', 101.0, 1001)");
+    }
+
+    @Test
+    public void testDynamicFilterEnabled__withPartitionPruningUsingDynamicFilter()
```

Comment:
nit: using single underscore
```suggestion
    public void testDynamicFilterEnabled_withPartitionPruningUsingDynamicFilter()
```

Reply:
Done